benchmarking/process_json_to_csv.jl

```diff
@@ -0,0 +1,171 @@
+  experiment_name::String
+  instance_name::String
+
+  # These fields correpond to those in ConvergenceInformation.
+  primal_objective::Float64
+  dual_objective::Float64
+  relative_optimality_gap::Float64
+  l2_primal_residual::Float64
+  l_inf_primal_residual::Float64
+  l2_dual_residual::Float64
+  l_inf_dual_residual::Float64
+  relative_l2_primal_residual::Float64
+  relative_l_inf_primal_residual::Float64
+  relative_l2_dual_residual::Float64
+  relative_l_inf_dual_residual::Float64
+  l_inf_primal_variable::Float64
+  l2_primal_variable::Float64
+  l_inf_dual_variable::Float64
+
+  termination_reason::FirstOrderLp.TerminationReason
+  iteration_count::Int32
+  cumulative_kkt_matrix_passes::Float64
+  solve_time_sec::Float64
+end
+
+function CsvRow()
+  return CsvRow(
+    "",  # experiment_name
+    "",  # instance_name
+    NaN,  # primal_objective
+    NaN,  # dual_objective
+    NaN,  # relative_optimality_gap
+    NaN,  # l2_primal_residual
+    NaN,  # l_inf_primal_residual
+    NaN,  # l2_dual_residual
+    NaN,  # l_inf_dual_residual
+    NaN,  # relative_l2_primal_residual
+    NaN,  # relative_l_inf_primal_residual
+    NaN,  # relative_l2_dual_residual
+    NaN,  # relative_l_inf_dual_residual
+    NaN,  # l_inf_primal_variable
+    NaN,  # l2_primal_variable
+    NaN,  # l_inf_dual_variable
+    FirstOrderLp.TERMINATION_REASON_UNSPECIFIED,  # termination_reason
+    0,  # iteration_count
+    NaN,  # cumulative_kkt_matrix_passes
+    NaN,  # solve_time_sec
+  )
+end
+
+function set_matching_fields(to, from)
+  for field_name in intersect(fieldnames(typeof(to)), fieldnames(typeof(from)))
+    setfield!(to, field_name, getfield(from, field_name))
+  end
+end
+
+function solve_log_to_csv_row(
+  log::FirstOrderLp.SolveLog,
+  experiment_name::String,
+)::CsvRow
+  row = CsvRow()
+  set_matching_fields(row, log)
+  row.experiment_name = experiment_name
+  row.cumulative_kkt_matrix_passes =
+    log.solution_stats.cumulative_kkt_matrix_passes
+
+  point_type = log.solution_type
+  # TODO: This doesn't properly handle the case of infeasibility certificates,
+  # whose stats are in log.solution_stats.infeasibility_information.
+  for convergence_information in log.solution_stats.convergence_information
+    if convergence_information.candidate_type == point_type
+      set_matching_fields(row, convergence_information)
+      break
+    end
+  end
+  return row
+end
+
+function rows_to_dataframe(rows::Vector{CsvRow})
+  return DataFrame(; StructArrays.components(StructArrays.StructArray(rows))...)
+end
+
+struct DatasetNameAndLocation
+  name::String
+  logs_directory::String
+end
+
+struct DatasetList
+  datasets::Vector{DatasetNameAndLocation}
+end
+
+StructTypes.StructType(::Type{DatasetNameAndLocation}) = StructTypes.Struct()
+StructTypes.StructType(::Type{DatasetList}) = StructTypes.Struct()
+
+function read_dataset(dataset_list::DatasetList)::DataFrame
+  rows = CsvRow[]
+  for name_and_location in dataset_list.datasets
```

Comment:
Optional: "name_and_location" -> "dataset" - the field name "datasets" suggests that it is a collection of dataset items.

Reply:
Done